exercises/series/src/lib.rs

```diff
@@ -0,0 +1,15 @@
+pub struct Series {}
```

Comment:
Why this interface? I'm not categorically rejecting it at this time, but I'd want a solid reason why the tested interface shouldn't be simply `pub fn series(digits: &str, len: usize) -> Option<Vec<String>>`.

Reply:
I think you‘re right. First I wanted to try another approach where I needed this struct, but then I rejected this and forgot to remove it.

Reply:
In fact, we don't even need the `Option`; we could have the function just return a 0-length `Vec` if the consumer requests a series of length greater than the input length, and lose no information.

Reply:
I like my approach with Result because this is one of the few rusty things in this exercise.

Reply:
Using `Result` to use more Rust features when it's not the natural interface is an antipattern.

There are plenty of exercises for which `Option` and `Result` make sense, and are implied by the requirements of the problem. This is not one of them: you know before ever entering the function exactly how many items you expect to get back from it.

Reply:
What would be expected if `len` = 0? A 0-length `Vec`?

Reply:
My instinct in that case would actually be to expect `vec!["".to_string(); digits.len() + 1]`.

Reasoning: in all other valid cases, you expect a `Vec` composed of `len`-length sequences, which contains `digits.len() + 1 - len`-length strings. Therefore, this sticks to the pattern.